source/ns_cmdline.c

```diff
@@ -41,6 +41,66 @@
 // force traces for this module
 //#define FEA_TRACE_SUPPORT
 
+// configurations
+/*
+#define MBED_CMDLINE_ENABLE_FEATURE_ALIASES 1
+#define MBED_CMDLINE_USE_DUMMY_SET_ECHO_COMMANDS 1
+#define MBED_CMDLINE_INIT_AUTOMATION_MODE 1
+#define MBED_CMDLINE_ENABLE_FEATURE_HISTORY 1
+#define MBED_CMDLINE_ENABLE_FEATURE_ESCAPE_HANDLING 1
+#define MBED_CMDLINE_ENABLE_FEATURE_OPERATORS 1
+#define MBED_CMDLINE_ENABLE_ALL_INTERNAL_COMMANDS 1
+#define MBED_CMDLINE_ENABLE_INTERNAL_VARIABLES 1
+#define MBED_CMDLINE_INCLUDE_MAN 1
+#define MBED_CMDLINE_MAX_LINE_LENGTH 100
+#define MBED_CMDLINE_ARGUMENTS_MAX_COUNT 2
+#define MBED_CMDLINE_HISTORY_MAX_COUNT 1
+*/
+
+#ifndef MBED_CMDLINE_USE_MINIMUM_SET
+#define MBED_CMDLINE_USE_MINIMUM_SET 0
+#endif
+
+#if MBED_CMDLINE_USE_MINIMUM_SET == 1
+// configure default minimum values, each value can be overwrite by compiling time
+#ifndef MBED_CMDLINE_ENABLE_FEATURE_ALIASES
```

Comment:
Should the `USE_MINIMUM_SET` override all others?

```
#if MBED_CMDLINE_USE_MINIMUM_SET
#undef MBED_CMDLINE_ENABLE_FEATURE_ALIASES
#undef MBED_CMDLINE_USE_DUMMY_SET_ECHO_COMMANDS
#undef MBED_CMDLINE_INIT_AUTOMATION_MODE
...
#define MBED_CMDLINE_ENABLE_FEATURE_ALIASES 0
#define MBED_CMDLINE_USE_DUMMY_SET_ECHO_COMMANDS 1
#define MBED_CMDLINE_INIT_AUTOMATION_MODE 1
...
#endif
```

Reply:
idea was that user can set still custom configurations even `MBED_CMDLINE_USE_MINIMUM_SET` is set - so custom would overwrite default minimum set, e.g.:
```
-DMBED_CMDLINE_USE_MINIMUM_SET=1 -DMBED_CMDLINE_MAX_LINE_LENGTH=200
```

does it make sense ?